### ISSUE TITLE
Support PHP 7.4 arrow functions

### ISSUE THREAD

**Issue by Th3Mouk** [kahlan/kahlan#336]
Hello 👋 !

PHP 7.4 will be released soon and currently arrow functions cause a 

```
message:`ParseError` Code(0) with message "syntax error, unexpected '=>' (T_DOUBLE_ARROW), expecting ')'"
```

To solve this, I propose to duplicate and adapt classic function specs and fixture in the project.
Can this case must be safely added to https://github.com/kahlan/kahlan/blob/master/src/Jit/Parser.php#L85 or do we need to check this apart with the PHP version @jails ? 

It's seems checking `token[0]` is not enough, we must check the if the `T_FUNCTION` is followed by `T_DOUBLE_ARROW` token too.

I can try an implementation this month if needed.

TY

**Comment by jails** [comment]
The parser is quite "basic". I had some plan to switch over [PHP-parser](https://github.com/nikic/PHP-Parser) but there was some constraints which made the transition impossible yet.

If arrow functions are identified by a `T_FUNCTION` followed by a  `T_DOUBLE_ARROW` and not only a `T_DOUBLE_ARROW`, maybe the support of arrow functions won't be that difficult. The only thing to do might be to make [_functionNode](https://github.com/kahlan/kahlan/blob/master/src/Jit/Parser.php#L476) to support both kind of functions.

Sure, feel free to give it a go and ask for help if you need some hints for this one, thank you !

**Comment by kieranajp** [comment]
Hi, I see that a PR was merged to support this last week :tada: 

Any chance of tagging a new release, please?

**Comment by samsonasik** [comment]
@kieranajp you can use `dev-dev` for now

**Comment by kieranajp** [comment]
Of course, I can lock to a commit hash too. I just prefer using a released version.

**Comment by jails** [comment]
Fixed in Kahkan 4.7
